channels/ON-CHAIN.md

```diff
@@ -0,0 +1,372 @@
+being able to withdraw locked coins might still be of use.
+
+The `from` account MUST be a participant in the target channel. The `amount`
+MUST be less or equal than the sum of all participants balances, i.e. channels
+cannot create coins out of thin air.
+
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| channel_id           | 32 |
+ ---------------------- ----
+| from                 | 32 |
+ ---------------------- ----
+| amount               | 32 |
+ ---------------------- ----
+| ttl                  | 2  |
+ ---------------------- ----
+| fee                  | 32 |
+ ---------------------- ----
+| nonce                | 32 |
+ ---------------------- ----
+```
+
+
+- `channel_id`: channel id as recorded on chain
+- `from`:
+- `amount`:
+- `ttl`:
+- `fee`:
+- `nonce`: the `from` account nonce
+
+
+(***TODO***: should a channel be considered closed if all the coins are taken from it?)
+
+## Closing channel on-chain
+
+### `channel_close_mutual`
+
+
+```
+  name                  size (bytes)
+ ---------------------- ----
+| channel_id           | 32 |
+ ---------------------- ----
+| initiator_amount     | 32 |
+ ---------------------- ----
+| responder_amount     | 32 |
+ ---------------------- ----
+| ttl                  | 2  |
+ ---------------------- ----
+| fee                  | 32 |
+ ---------------------- ----
+| nonce                | 32 |
+ ---------------------- ----
+```
+
+
+- `channel_id`: channel id as recorded on chain
+- `initiator_amount`: final balance for the initiator
+- `responder_amount`: final balance for the responder
+- `ttl`:
+- `fee`:
+- `nonce`: taken from the `initiator` account
+
+`initiator_amount` and `responder_amount` are the agreed upon distribution of
+coins. To get the final outcome of the channel, the fees have to get accounted
+for. It is advised that peers choose an even amount, otherwise one peer will end
+up paying more, although this should not be an issue in practice.
+
+```
+if initiator_amount + responder_amount < fee
+  return error
+else if initiator_amount >= ceil(fee/2) && responder_amount >= floor(fee/2)
+  initiator_final := initiator_amount - ceil(fee/2)
+  responder_final := responder_amount - floor(fee/2)
+else if responder_amount >= ceil(fee/2) && initiator_amount >= floor(fee/2)
+  responder_final := responder_amount - ceil(fee/2)
+  initiator_final := initiator_amount - floor(fee/2)
+else if initiator_amount > responder_amount
+  initiator_final := initiator_amount - fee + responder_amount
+  responder_final := 0.0
+else
+  responder_final := responder_amount - fee + initiator_amount
+  initiator_final := 0.0
+```
+
+
+
+#### Requirements
+
+This transaction MUST have valid signatures of all involved parties.
+
+After this transaction has been included in a block, the channel MUST be
+considered closed and allow no further modifications.
+
+
+### `channel_close_solo`
+
+In order to close a channel unilaterally, a user has to send a
```

Comment:
I would expect something like "The `from` account MUST be a participant in the target channel." - not any arbitrary user.